src/main/java/at/hannibal2/skyhanni/data/BitsAPI.kt

```diff
@@ -0,0 +1,200 @@
+    )
+
+    // GUI patterns
+    private val bitsGuiGroup = bitsDataGroup.group("gui")
+
+    private val bitsAvailableMenuPattern by bitsGuiGroup.pattern(
+        "availablemenu",
+        "§7Bits Available: §b(?<toClaim>[\\w,]+)(§3.+)?"
+    )
+
+    private val fameRankSbMenuPattern by bitsGuiGroup.pattern(
+        "sbmenufamerank",
+        "§7Your rank: §e(?<rank>.*)"
+    )
+
+    private val fameRankCommunityShopPattern by bitsGuiGroup.pattern(
+        "communityshopfamerank",
+        "§7Fame Rank: §e(?<rank>.*)"
+    )
+
+    private val bitsGuiNamePattern by bitsGuiGroup.pattern(
+        "mainmenuname",
+        "^SkyBlock Menu$"
+    )
+
+    private val bitsGuiStackPattern by bitsGuiGroup.pattern(
+        "mainmenustack",
+        "^§6Booster Cookie$"
+    )
+
+    private val fameRankGuiNamePattern by bitsGuiGroup.pattern(
+        "famerankmenuname",
+        "^(Community Shop|Booster Cookie)$"
+    )
+
+    private val fameRankGuiStackPattern by bitsGuiGroup.pattern(
+        "famerankmenustack",
+        "^(§aCommunity Shop|§eFame Rank)$"
+    )
+
+    @SubscribeEvent
+    fun onScoreboardChange(event: ScoreboardChangeEvent) {
+        if (!isEnabled()) return
+        for (line in event.newList) {
+            val message = line.trimWhiteSpace().removeResets()
+
+            bitsScoreboardPattern.matchMatcher(message) {
+                val amount = group("amount").formatInt()
+
+                if (amount > bits) {
+                    bitsToClaim -= amount - bits
+                    ChatUtils.debug("You have gained §3${amount - bits} Bits §7according to the scoreboard!")
+                }
+                bits = amount
+
+                return
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!isEnabled()) return
+        val message = event.message.trimWhiteSpace().removeResets()
+
+        bitsFromFameRankUpChatPattern.matchMatcher(message) {
+            val amount = group("amount").formatInt()
+            bitsToClaim += amount
+
+            return
+        }
+
+        boosterCookieAte.matchMatcher(message) {
+            bitsToClaim += (defaultcookiebits * (currentFameRank?.bitsMultiplier ?: return)).toInt()
+
+            return
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryFullyLoaded(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+
+        val stacks = event.inventoryItems
+
+        if (bitsGuiNamePattern.matches(event.inventoryName)) {
+            val cookieStack = stacks.values.lastOrNull { bitsGuiStackPattern.matches(it.displayName) }
+            if (cookieStack != null) {
+                for (line in cookieStack.getLore()) {
+                    bitsAvailableMenuPattern.matchMatcher(line) {
+                        bitsToClaim = group("toClaim").formatInt()
+
+                        return
+                    }
+                }
+            }
+            return
```

Comment:
Less indent (sorry for the bad formatting)
```suggestion
            val cookieStack = stacks.values.lastOrNull { bitsGuiStackPattern.matches(it.displayName) } ?: return
                for (line in cookieStack.getLore()) {
                    bitsAvailableMenuPattern.matchMatcher(line) {
                        bitsToClaim = group("toClaim").formatInt()

                        return
                    }
                }
            return
```